src/software/estop/threaded_estop_reader.cpp

```diff
@@ -0,0 +1,89 @@
+
+#include "threaded_estop_reader.h"
+
+#include <boost/bind/bind.hpp>
+#include <boost/function.hpp>
+#include <cstdio>
+#include <cstring>
+#include <thread>
+#include <utility>
+
+#include "software/logger/logger.h"
+
+
+ThreadedEstopReader::ThreadedEstopReader(int startup_ms, int interval_ms,
+                                         std::unique_ptr<UartCommunication> uart_reader)
+    : startup_ms(startup_ms),
+      interval_ms(interval_ms),
+      timer(io_service, this->interval_ms),
+      estop_msg(0),
```

Comment:
does this need to be an instance variable?

Reply:
nope, removed